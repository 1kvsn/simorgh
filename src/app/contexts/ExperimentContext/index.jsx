import React, { createContext } from 'react';


const ExperimentContext = React.createContext({});

const ExperimentContextProvider = ({ children, dataMvtValue }) => {
  return (
    <ExperimentContext.Provider value={dataMvtValue}>
      {children}
    </ExperimentContext.Provider>
  );
};

const ExperimentContextConsumer = ExperimentContext.Consumer;

export { ExperimentContext, ExperimentContextProvider, ExperimentContextConsumer };
