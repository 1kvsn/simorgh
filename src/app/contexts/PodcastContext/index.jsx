import React from 'react';
import { node, string } from 'prop-types';
import podcastData from '#app/lib/config/podcastExternalLinks/loadableConfig';
import variantPropType from '../../models/propTypes/variants';
import { getVariant } from '#lib/utilities/variantHandler';

export const PodcastContext = React.createContext({});

export const PodcastContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = podcastData[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={PodcastContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

PodcastContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

PodcastContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
