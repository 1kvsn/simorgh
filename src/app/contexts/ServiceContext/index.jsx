import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import variantPropType from '../../models/propTypes/variants';
import { getVariant } from '#lib/utilities/variantHandler';
import getLangOverride from '#lib/utilities/langHandler';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({
  children,
  service,
  variant,
  pageLang,
}) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getLangOverride(pageLang) || getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
  pageLang: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
  pageLang: null,
};
