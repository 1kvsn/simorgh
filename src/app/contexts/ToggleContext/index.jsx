import React, { createContext, useReducer } from 'react';
import { node, object } from 'prop-types';
import { toggleReducer } from './reducer';

const ToggleContext = createContext({});

const ToggleContextProvider = ({ children, toggles }) => {
  const [toggleState, toggleDispatch] = useReducer(toggleReducer, toggles);

  return (
    <ToggleContext.Provider value={{ toggleState, toggleDispatch }}>
      {children}
    </ToggleContext.Provider>
  );
};

const ToggleContextConsumer = ToggleContext.Consumer;

ToggleContextProvider.propTypes = {
  children: node.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  toggles: object.isRequired,
};

export { ToggleContext, ToggleContextProvider, ToggleContextConsumer };
