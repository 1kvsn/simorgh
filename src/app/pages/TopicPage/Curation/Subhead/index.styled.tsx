import styled from '@emotion/styled';

import { getDoublePica } from '../../../../legacy/psammead/gel-foundations/src/typography';

import { getSansBold } from '../../../../legacy/psammead/psammead-styles/src/font-styles';
import {
  C_POSTBOX,
  C_GREY_10,
} from '../../../../legacy/psammead/psammead-styles/src/colours';

export default styled.a<{ service: string; script: string }>`
  ${({ service }) => getSansBold(service)}
  ${({ script }) => getDoublePica(script)}
  color: ${C_GREY_10};
  text-decoration: none;
  display: inline-block;

  span {
    display: inline-block;
    position: relative;
  }
  &:hover,
  &:focus {
    color: ${C_POSTBOX};
    span::after {
      bottom: ${3 / 16}rem;
      content: '';
      background: ${C_POSTBOX};
      width: 100%;
      height: ${2 / 16}rem;
      position: absolute;
      left: 0;
    }
  }
  svg {
    margin-inline-start: 0.5rem;
    fill: currentColor;
    height: ${7 / 8}rem;
    position: relative;
  }
`;
