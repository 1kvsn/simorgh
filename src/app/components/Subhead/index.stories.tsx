import React from 'react';

import { withServicesKnob } from '#psammead/psammead-storybook-helpers/src';
import { withKnobs } from '@storybook/addon-knobs';
import { ServiceContextProvider } from '#contexts/ServiceContext';

import Subheading from '.';



// @ts-ignore
const Component = ({ service, variant }) => {
  return (
    <ServiceContextProvider service={service} variant={variant}>
        <Subheading children="These are the children"/>
    </ServiceContextProvider>
  );
};


export default {
  title: 'Topic/Curations/Subheading - Normal',
  Component,
  decorators: [withKnobs, withServicesKnob()],
  parameters: { chromatic: { disable: true } },
};

export const Example = Component;

