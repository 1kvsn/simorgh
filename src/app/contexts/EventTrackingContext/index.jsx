import React, { createContext, useContext } from 'react';
import { node } from 'prop-types';

import { pageDataPropType } from '#models/propTypes/data';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContext } from '#contexts/RequestContext';
import { buildATIClickParams } from '#containers/ATIAnalytics/params';
import useToggle from '#hooks/useToggle';
import {
  ARTICLE_PAGE,
  FRONT_PAGE,
  MEDIA_PAGE,
  MOST_READ_PAGE,
  MOST_WATCHED_PAGE,
  INDEX_PAGE,
  FEATURE_INDEX_PAGE,
  MEDIA_ASSET_PAGE,
  STORY_PAGE,
  PHOTO_GALLERY_PAGE,
  CORRESPONDENT_STORY_PAGE,
} from '#app/routes/utils/pageTypes';

export const EventTrackingContext = createContext({});

export const EventTrackingContextProvider = ({ children, pageData }) => {
  let pageIdentifier;
  let platform;
  let statsDestination;

  const requestContext = useContext(RequestContext);
  const serviceContext = useContext(ServiceContext);
  const { enabled: eventTrackingIsEnabled } = useToggle('eventTracking');

  if (pageData && eventTrackingIsEnabled) {
    try {
      ({ pageIdentifier, platform, statsDestination } = buildATIClickParams(
        pageData,
        requestContext,
        serviceContext,
      ));
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(
        `ATI Event Tracking Error: Could not parse tracking values from page data:\n${error.message}`,
      );
    }
  }

  const { pageType } = requestContext;
  const campaignName = {
    [ARTICLE_PAGE]: 'article',
    [FRONT_PAGE]: 'index-home',
    [MEDIA_PAGE]: 'player-episode-tv',
    [MOST_READ_PAGE]: 'list-datadriven-read',
    [MOST_WATCHED_PAGE]: 'list-datadriven-watched',
    [INDEX_PAGE]: 'index-section',
    [FEATURE_INDEX_PAGE]: 'index-section-fix',
    [MEDIA_ASSET_PAGE]: 'article-media-asset',
    [STORY_PAGE]: 'article-sty',
    [PHOTO_GALLERY_PAGE]: 'article-photo-gallery',
    [CORRESPONDENT_STORY_PAGE]: 'article-csp',
  }[pageType];

  if (!campaignName && pageData) {
    // eslint-disable-next-line no-console
    console.error(
      `ATI Event Tracking Error: Could not get the page type's campaign name`,
    );
  }

  const hasRequiredProps = [
    campaignName,
    pageIdentifier,
    platform,
    statsDestination,
  ].every(Boolean);
  const trackingData = hasRequiredProps
    ? {
        campaignName,
        pageIdentifier,
        platform,
        statsDestination,
      }
    : {};

  return (
    <EventTrackingContext.Provider value={trackingData}>
      {children}
    </EventTrackingContext.Provider>
  );
};

EventTrackingContextProvider.propTypes = {
  children: node.isRequired,
  pageData: pageDataPropType,
};

EventTrackingContextProvider.defaultProps = {
  pageData: null,
};
