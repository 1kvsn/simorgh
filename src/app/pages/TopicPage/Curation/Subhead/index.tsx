import React, { useContext } from 'react';

import { H2 } from './index.styled';

import { RightChevron } from '../../icons';

import { ServiceContext } from '../../../../contexts/ServiceContext';

interface Props {
  children: React.ReactNode;
  href?: string;
}

const Subhead = ({ children, href }: Props) => {
  const { service, script } = useContext(ServiceContext) as {
    script: string;
    service: string;
  };
  const Wrapper = href
    ? ({ children: innerChildren }) => (
        <a href={href}>
          {innerChildren} <RightChevron />
        </a>
      )
    : React.Fragment;
  return (
    <H2 service={service} script={script} id="content" tabIndex={-1}>
      <Wrapper>{children}</Wrapper>
    </H2>
  );
};

Subhead.defaultProps = { href: '' };

export default Subhead;
