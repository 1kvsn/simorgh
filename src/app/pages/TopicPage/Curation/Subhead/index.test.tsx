import React from 'react';
import { render } from '@testing-library/react';
import { ServiceContextProvider } from '../../../../contexts/ServiceContext';
import Subhead from '.';

/* eslint-disable react/prop-types */
const SubheadWithContext = ({
  children = '',
  href = '',
  service = 'mundo',
} = {}) => (
  <ServiceContextProvider service={service}>
    <Subhead href={href}>{children}</Subhead>
  </ServiceContextProvider>
);

describe('A11y', () => {
  it('should render a link correctly with the url contained in the href', () => {
    render(<SubheadWithContext>My Text</SubheadWithContext>);

    const headingElement = document.querySelector('h2');
    expect(headingElement).toMatchSnapshot();
    expect(headingElement.innerHTML).toBe('My Text');
  });

  it('should render children within an h2', () => {});
});
