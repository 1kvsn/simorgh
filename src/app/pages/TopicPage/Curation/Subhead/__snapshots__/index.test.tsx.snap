// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A11y should render a link correctly with the url contained in the href 1`] = `
.emotion-0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #141414;
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

<h2
  class="emotion-0 emotion-1"
  id="content"
  tabindex="-1"
>
  My Text
</h2>
`;
